service/service.ml

```diff
@@ -59,15 +59,19 @@ module Make (Opam_repo : Opam_repository_intf.S) = struct
       let hash = Store.Hash.of_hex commit.Remote_commit.hash in
       Opam_repo.open_store ~repo_url () >>= fun store ->
       Store.mem store hash >>= function
-      | true -> Lwt.return_unit
+      | true -> Opam_repo.close_store store
       | false -> (
+          Opam_repo.close_store store >>= fun () ->
           Fmt.pr "Need to update %s to get new commit %a@." repo_url
             Store.Hash.pp hash;
           Opam_repo.fetch ~repo_url () >>= fun () ->
           Opam_repo.open_store ~repo_url () >>= fun new_store ->
           Store.mem new_store hash >>= function
-          | false -> Fmt.failwith "Still missing commit after update!"
-          | true -> Lwt.return_unit)
+          | false ->
+              Opam_repo.close_store new_store >>= fun () ->
+              Fmt.failwith "Still missing commit after update!"
+          | true -> Opam_repo.close_store new_store)
+    (*Closing the store after usage is necessary to prevent file descriptor leaks*)
```

Comment:
You can factor the `close_store` call.

```suggestion
          (* Closing the store after usage is necessary to prevent file descriptor leaks. *)
          Opam_repo.close_store new_store >>= fun () ->
          if r then Lwt.return_unit
          else Fmt.failwith "Still missing commit after update!")
```